screens/SearchResultsScreen.js

```diff
@@ -1,29 +1,89 @@
-  const [businesses, setBusinesses] = useState([])
+  const { searchInput } = route.params;
+  const [businesses, setBusinesses] = useState([]);
+  const [filteredBusinesses, setFilteredBusinesses] = useState([]);
+
+  const categoryData = [
+    { label: "Hospitality", value: "hospitality" },
+    { label: "Retail", value: "retail" },
+    { label: "Health and Wellbeing", value: "health-wellbeing" },
+    { label: "Services", value: "services" },
+    { label: "Other", value: "other" }
+  ];
+
+  const nearestToMeData = [{ label: "Date added", value: "date-added" }];
 
   useEffect(() => {
     const fetchBusinesses = async () => {
-      const results = await fetch(`https://carrythroughcovid.herokuapp.com/api/v1/business?input=${searchInput}`)
-      const parsed = await results.json()
-      setBusinesses(parsed)
-    }
-    fetchBusinesses()
-  }, [setBusinesses])
+      const results = await fetch(
+        `https://carrythroughcovid.herokuapp.com/api/v1/business?input=${searchInput}`
+      );
+      const parsed = await results.json();
+      setBusinesses(parsed);
+      setFilteredBusinesses(parsed);
+    };
+    fetchBusinesses();
+  }, [setBusinesses]);
+
+  const handleCategoryChange = selectedCategory => {
+    const filtered = businesses.filter(
+      business => business.categories[0].name == selectedCategory
+    );
+    setFilteredBusinesses(filtered);
+  };
 
   return (
     <ScrollView style={styles.container}>
       <View style={styles.paddingContainer}>
         <Input placeholder="Search for a business" />
-        <Text style={styles.resultsText}>{businesses.length} results</Text>
-        {businesses.length > 0 &&
-          businesses.map(business => (
+        <View style={styles.dropDownContainer}>
+          <View style={styles.dropDown}>
+            <RNPickerSelect
+              style={pickerSelectStyles}
+              onValueChange={e => handleCategoryChange(e)}
+              items={categoryData}
+              placeholder={{ label: "All categories", value: null }}
+              placeholderTextColor="#3F3356"
+              Icon={() => {
+                return <FontAwesomeIcon icon={faSortDown} color={"#3F3356"} />;
+              }}
+            />
+          </View>
+          <View style={styles.dropDown}>
+            <RNPickerSelect
+              style={pickerSelectStyles}
+              onValueChange={value => console.log(value)}
+              items={nearestToMeData}
+              placeholder={{ label: "Nearest to me", value: "nearest-to-me" }}
+              placeholderTextColor="#3F3356"
+              Icon={() => {
+                return <FontAwesomeIcon icon={faSortDown} color={"#3F3356"} />;
+              }}
+            />
+          </View>
+        </View>
+        <Text style={styles.resultsText}>
+          {filteredBusinesses.length}{" "}
+          {filteredBusinesses.length === 1 ? "result" : "results"}
+        </Text>
+        {filteredBusinesses.length > 0 &&
```

Comment:
Re: the callout about "All Categories" - could make this a ternary, display `businesses` if `filteredBusinesses.length < 0`, and set `filteredBusinesses` to empty in the `onChange` if "All categories" is selected.

Reply:
Good call 👍

Another option I thought of mid-breakfast was doing a check in `handleCategoryChange()` to see if there is actually user selection. If it's `""` then we return everything.

I'll do some testing later today. 😄

Reply:
I've updated `handleCategoryChange()` as above and it's working as expected 🙂